### PR TITLE
Transport parameters are mandatory. Fixes #2528

### LINKED ISSUE #2528: Are transport parameters mandatory?
All of them have defaults.

### PR DESCRIPTION
**Pull request by ekr**
Fixes #2528

### CODE REVIEW

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -1249,7 +1249,10 @@ quic_transport_parameters extension carries a TransportParameters struct when
 the version of QUIC defined in {{QUIC-TRANSPORT}} is used.
 
 The quic_transport_parameters extension is carried in the ClientHello and the
-EncryptedExtensions messages during the handshake.
+EncryptedExtensions messages during the handshake. Endpoints MUST send the
+quic_transport_parameters extension; endpoints which receive ClientHello
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md]:
```suggestion
quic_transport_parameters extension; endpoints that receive ClientHello
```
CMOS

File: draft-ietf-quic-tls.md [hunk 2]
```diff
@@ -1249,7 +1249,10 @@ quic_transport_parameters extension carries a TransportParameters struct when
...
+or EncryptedExtensions messages without the transport_parameters extension MUST
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md]:
```suggestion
or EncryptedExtensions messages without the quic_transport_parameters extension MUST
```

File: draft-ietf-quic-tls.md [hunk 3]
```diff
@@ -1249,7 +1249,10 @@ quic_transport_parameters extension carries a TransportParameters struct when
...
+terminate the TLS handshake with a fatal missing_extension alert.
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md]:
```suggestion
terminate the TLS handshake with a fatal missing_extension alert (an error of 0x16d).
```

File: draft-ietf-quic-tls.md [hunk 4]
```diff
@@ -1249,7 +1249,11 @@ quic_transport_parameters extension carries a TransportParameters struct when
 the version of QUIC defined in {{QUIC-TRANSPORT}} is used.
 
 The quic_transport_parameters extension is carried in the ClientHello and the
-EncryptedExtensions messages during the handshake.
+EncryptedExtensions messages during the handshake. Endpoints MUST send the
+quic_transport_parameters extension; endpoints that receive ClientHello or
+EncryptedExtensions messages without the quic_transport_parameters extension
+MUST terminate the TLS handshake with a fatal missing_extension alert (an error
```
Comment (janaiyengar) [inline draft-ietf-quic-tls.md]:
```suggestion
MUST close the connection with an error of type 0x16d (equivalent to a fatal TLS missing_extension alert, see Section {{tls-errors}}).
```

File: draft-ietf-quic-tls.md [hunk 5]
```diff
@@ -1258,8 +1262,8 @@ parameters will cause the handshake to fail.
 
 Endpoints MUST NOT send this extension in a TLS connection that does not use
 QUIC (such as the use of TLS with TCP defined in {{!TLS13}}).  A fatal
-unsupported_extension alert MUST be sent if this extension is received when the
-transport is not QUIC.
+unsupported_extension alert (an error of 0x16e) MUST be sent if this extension
+is received when the transport is not QUIC.
```
Comment (janaiyengar) [inline draft-ietf-quic-tls.md]: This is not just in your text, but while you're here, also replace the text with:
"The connection MUST be closed with an error code of type 0x16e (equivalent to a fatal TLS unsupported_extension alert, see Section {{tls-errors}}) if the transport is not QUIC."

File: draft-ietf-quic-tls.md (hunk 4 above)
Comment (MikeBishop) [reply draft-ietf-quic-tls.md]: This will result in "Section Section X.Y" in the final output.

File: draft-ietf-quic-tls.md (hunk 5 above)
Comment (MikeBishop) [reply draft-ietf-quic-tls.md]: Same.

Comment (nharper) [reply draft-ietf-quic-tls.md]: I agree with @MikeBishop. Specifying the QUIC error code to send if the extension is received when the transport isn't QUIC doesn't make much sense.

File: draft-ietf-quic-tls.md (hunk 4 above)
Comment (MikeBishop) [reply draft-ietf-quic-tls.md]: Also, that comma should be a semicolon; see #2626.

**Pull request merged**